extension.js

```diff
@@ -36,6 +36,10 @@ class ExtensionDBusInterface {
             return;
 
         Main.wm.skipNextEffect(current_window.get_compositor_private());
+        if (settings.get_boolean('window-maximize')) {
+            settings.set_boolean('window-maximize', false);
+            settings.set_double('window-height', 1.0);
```

Comment:
Changing `window-height` will trigger `update_window_height`. Did the other approach (`move_resize_window()` to the entire `workarea`) not work?

And, by the way, resizing should be done after `.unmaximize()` call, I think

Reply:
>Changing window-height will trigger update_window_height

You're right! Then it makes more sense to just call `update_window_height()` in this situaton

> Did the other approach (move_resize_window() to the entire workarea) not work?

That's no different from changing the height to 100% right?

> And, by the way, resizing should be done after .unmaximize() call, I think

I think I tried this and the order mattered.